src/components/Pagination.js

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useState } from "react";
+import { uniq } from "lodash-es";
+
+export const Pagination = ({
+  total,
+  currentPage,
+  onPageChange,
+  nextPage,
+  previousPage,
+  postsPerPage,
+  handleSelectChange,
+}) => {
+  const [pagesToShow, setPagesToShow] = useState([1]);
+
+  const lastPage = Math.ceil(total / postsPerPage);
+
+  const getPagesToShow = () => {
+    if (lastPage === 1) {
+      setPagesToShow([1]);
+    } else {
+      const tempNumbers = [1, lastPage];
+
+      const firstValue = currentPage === 1 ? currentPage + 1 : 0;
+      const secondValue = currentPage - 1;
+      const thirdValue = currentPage;
```

Comment:
This variable is not needed. Use currentPage directly

Reply:
removed.